homeassistant/components/rainbird/switch.py

```diff
@@ -57,49 +70,63 @@ def setup_platform(
             name = zone_config.get(CONF_FRIENDLY_NAME)
             devices.append(
                 RainBirdSwitch(
+                    coordinator,
                     controller,
                     zone,
                     time,
                     name if name else f"Sprinkler {zone}",
                 )
             )
 
-    add_entities(devices, True)
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except ConfigEntryNotReady as err:
+        raise PlatformNotReady(f"Failed to load zone state: {str(err)}") from err
 
-    def start_irrigation(service: ServiceCall) -> None:
+    async_add_entities(devices)
+
+    async def start_irrigation(service: ServiceCall) -> None:
         entity_id = service.data[ATTR_ENTITY_ID]
         duration = service.data[ATTR_DURATION]
 
         for device in devices:
             if device.entity_id == entity_id:
-                device.turn_on(duration=duration)
+                await device.async_turn_on(duration=duration)
```

Comment:
Probably break out the duration to a number entity in the future and just have the switch turn the irrigation on/off.

Reply:
Do you think it makes sense to have one per sprinkler pus a global value? I also wonder if the per sprinkler numbers should be disabled by default.

I was considering this think this made more sense than a large table in the options flow.

Reply:
Maybe. How do you switch between the global value and the individual value? Disable the individual entity?

Reply:
Yeah...

Another option is to deprecate support per sprinkler defaults and preserving as service params.

In the device, the irrigation time for each sprinkler is specified as part of a program. Isy also look closer at the API and see if there are more ways to expose the existing programs rather than.

Reply:
Is it OK for a NumberEntity to store configuration state? I wasn't sure that it was generally agreed to be an OK use case based on arch discussions in the last year and a half

Reply:
Can you give an example?

Reply:
OK so my impression here is that what you're saying is let's try to express this in simpler components that can be composed together.
1) we don't need a "start irrigation service", and we can just call the service to turn the irrigation switch on and everything is fine.
2) the irrigation duration can be a number and allow input from the UI. it can be a `config` entity too. (let's ignore per-zone durations for now)

When running this sprinkler, you send a message to the device of a zone and a duration. The duration itself is not stored on the device anywhere for that. 

So a Number entity for duration in this case would need to store its state only persisted in the state machine and restored on startup. It may be initially populated from yaml import or using the existing default value of 6 minutes.

My impression is that number entities typically represent some config of an external service, not their own config values., but i may be wrong about this. I couldn't see prior art doing what i described. The alternative would be configuration options.

Reply:
I don't believe we allow virtual settings like that yet. Then my suggestion is not good, I think. Maybe a custom service is better?